data/maps/Route8/scripts.pory

```diff
@@ -1,12 +1,392 @@
 raw `
-Route8_MapScripts::
-	.byte 0
+
 
 Route8_EventScript_UndergroundPathSign::
 	msgbox Route8_Text_UndergroundPathSign, MSGBOX_SIGN
 	end
 `
+
+
+const GROWLITHE = 19
+mapscripts Route8_MapScripts {
+	MAP_SCRIPT_ON_LOAD {
+		if(flag(FLAG_CSR_ROUTE8_PUSHED_BENCH)) {
+			setbench(BENCH_ROUTE8_AFTER)
+			setmetatile(35, 2, 0x3C0, FALSE)
+			setmetatile(36, 2, 0x3C1, FALSE)
+			setmetatile(37, 2, 0x3C2, FALSE)
+		} else {
+			setbench(BENCH_ROUTE8_BEFORE)
+		}
+	}
+}
+
+
+
 script Route8_RouteSign {
 	msgbox("ROUTE 8\nNEW WILD POKéMON: 1", MSGBOX_SIGN)
 	end
 }
+
+script Route8_StartSnowing {
+	setweather(WEATHER_SNOW)
+	doweather
+	release
+}
+
+script Route8_StartSunning {
+	setweather(WEATHER_SUNNY_CLOUDS)
+	doweather
+	release
+}
+
+script Route8_StartSandstorm {
+	setweather(WEATHER_SANDSTORM)
+	doweather
+	release
+}
+
+script Route8_StartRaining {
+	setweather(WEATHER_RAIN_THUNDERSTORM)
+	doweather
+	release
+}
+
+script Route8_DrySoil {
+	msgbox("The soil is intensely dry and arid.\nIt hasn't been moistened in ages.\pAttempting to plant a berry in such\nsere land would be a waste.", MSGBOX_DEFAULT)
+	end
+}
+
+script Route8_DrowningFriends {
+	if(flag(FLAG_CSR_GOT_PAY_DAY)) {
+		msgbox("…\pAll my friends are dead.\n", MSGBOX_NPC)
```

Comment:
\n at the end of the text statement? 🤔  Not a problem, but not needed I would assume

Reply:
Oh yea my text formatting program does that sometimes. I should fix that bug at some point